play-json/jvm/src/test/scala/play/api/libs/json/JsonSpec.scala

```diff
@@ -485,16 +498,28 @@ class JsonSpec extends org.specs2.mutable.Specification {
         "key4" -> Json.arr(1, 2.5, "value2", false, JsNull),
         "key5" -> Json.obj(
           "key6" -> "こんにちは",
-          "key7" -> BigDecimal("12345678901234567890.123456789")
-        )
+          "key7" -> BigDecimal("12345678901234567890.123456789"),
+        ),
       )
       def stream = new java.io.ByteArrayInputStream(
-        js.toString.getBytes("UTF-8")
+        js.toString.getBytes("UTF-8"),
       )
 
       Json.parse(stream).mustEqual(js)
     }
 
+    "parse large strings with higher constraint" in withJsonConfig(allowLargeStrings) { () =>
+      val builder = new StringBuilder()
+      1.to(30000000).foreach { _ => builder += 'a' }
+      val original = Json.obj(
+        "key" -> builder.result(),
+      )
+
+      val str = Json.stringify(original)
+
+      Json.parse(str).mustEqual(original)
+    }
+
```

Comment:
@tmccombs If you think it's still useful, please provide a pull request with this test against the main branch, thanks!